content/blog/2021/2021-02-10-explaining-models.md

```diff
@@ -0,0 +1,53 @@
++++
+author = "Ajda Pretnar"
+date = "2021-02-10"
+draft = false
+title = "Explaining Predictive Models"
+type = "blog"
+thumbImage = "/blog_img/2021/2021-02-10-explain-models-small.png"
+frontPageImage = "/blog_img/2021/2021-02-10-explain-models-small.png"
+blog = ["model explanation", "explainable AI", "explain", "predictive modelling"]
+shortExcerpt = "New Orange add-on for explaining predictive models."
+longExcerpt = "New Orange Explain add-on for understanding predictions and predictive models."
+x2images = false  # true if using retina screenshots, else false
++++
+
+It is easy to build powerful predictive models in Orange. But how does the model "look like"? Which attributes and which values of those attributes are important? And when making predictions, which attributes contributed to the decision? **Orange's new Explain add-on** helps you answer all those questions.
+
+Related: {{< link_new url="/blog/2019/2019-11-20-belgrade-workshop/" name="Explaining Models">}}
+
+Go to **Options --> Add-ons** and install Explain add-on. Restart Orange for the add-on to appear. It only contains two widgets, but boy are they great!
+
+Let us start with the attrition data set from the Datasets widget. We will go with **Attrition - Train**, which a data set on which employees resigned from the company and which stayed. The target variable is called Attrition, where **No** means that the employee stayed and **Yes** that the employee resigned. The other attributes describe the employee - her position, education, department, years since promotion, and so on.
+
+{{% workflow-screenshot src="/blog_img/2021/2021-02-10-data-table.png" %}}
+
+Next, we will build a simple logistic regression predictive model. If inspecting the model in Test and Score, we learn that the model has an AUC of 0.788 and 86 % CA. But what *kind* of a model is this? How does it makes its decisions?
+
+Let us add Explain Model to Logistic Regression and add another connection passing the data. The workflow should look like this:
+
+{{% workflow-screenshot src="/blog_img/2021/2021-02-10-model-workflow.png" %}}
+
+Now open Explain Model. The widget lists top ranked variables, which means they contribute the most to the selected target variable. As we are trying to understand why people leave the company, we have set the target variable to Yes.
+
+The highest ranked variable is OverTime - this is the variable with the highest impact on the prediction. Having a high value of overtime (red dots on the right) means the employee is likely to quit. Also, having low job satisfaction contributes to attrition (blue values on the right). The visualization shows the values that have a high impact on the model on the right and those with a low impact on the left. The color of dot represents the value of the attribute (red for higher values and blue for lower).
```

Comment:
OverTime is a discrete variable and it has two values Yes or No. The logistic regression model first continuize the discrete variables. In the case of OverTime it gets two columns OverTime - Yes, Overtime - No. Because of this behavior, it is wrong to say `Having a high value of overtime`, but on the other side, it is difficult to explain such that it would be simple for a user. 

Maybe we can say something like `Having a value Yes in Categorical attribute OverTime (red dots on the right) means the employee is likely to quit`.

Reply:
Values on the right have a high impact on the prediction of the selected class and values left from the center have an impact too, on prediction against this class. So I would change this sentence `The visualization shows the values that have a high impact on the model on the right and those with a low impact on the left` -> `The visualization shows the values that have a high impact on the prediction of selected class on the right and those which vote against the selected class on the left.`